micromasters/serializers.py

```diff
@@ -0,0 +1,78 @@
+"""
+Serializers for Django contrib models
+"""
+import logging
+from rest_framework import serializers
+from django.contrib.auth.models import User, AnonymousUser
+from django.core.exceptions import ObjectDoesNotExist
+from rolepermissions.verifications import has_role
+from roles.models import Instructor, Staff
+from backends.edxorg import EdxOrgOAuth2
+
+log = logging.getLogger(__name__)
+
+
+class UserSerializer(serializers.ModelSerializer):
+    """Serializer for users."""
+    first_name = serializers.SerializerMethodField()
+    last_name = serializers.SerializerMethodField()
+    preferred_name = serializers.SerializerMethodField()
+    is_staff = serializers.SerializerMethodField()
+    social_username = serializers.SerializerMethodField()
+
+    class Meta:
+        model = User
+        fields = (
+            "username", "email", "social_username",
+            "first_name", "last_name", "preferred_name",
+            "is_staff",
+        )
+
+    def get_first_name(self, obj):  # pylint: disable=no-self-use
+        "Get first_name from user profile"
+        if obj.profile:
+            return obj.profile.first_name
+        return None
+
+    def get_last_name(self, obj):  # pylint: disable=no-self-use
+        "Get last_name from user profile"
+        if obj.profile:
+            return obj.profile.last_name
+        return None
+
+    def get_preferred_name(self, obj):  # pylint: disable=no-self-use
+        "Get preferred_name from user profile"
+        if obj.profile:
+            return obj.profile.preferred_name
+        return None
+
+    def get_is_staff(self, obj):  # pylint: disable=no-self-use
```

Comment:
We shouldn't conflate staff and instructor here. Can you change this to return the roles assigned to the user instead, or remove it since it's not being used? We currently output roles information to `SETTINGS` here, thought this isn't done on the program pages yet: https://github.com/mitodl/micromasters/blob/master/ui/views.py#L53